internal/db/controllerproxy.go

```diff
@@ -1,36 +1,76 @@
+// Copyright 2025 Canonical.
+
 package db
 
 import (
 	"context"
+	"encoding/json"
 
 	"github.com/canonical/jimm/v3/internal/dbmodel"
 	"github.com/canonical/jimm/v3/internal/errors"
 	"github.com/canonical/jimm/v3/internal/servermon"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+)
+
+const (
+	// These constant are used to create the appropriate identifiers for controller proxy data.
+	controllerProxyKind = "controllerProxy"
+	controllerProxyTag  = "controllerProxy"
+	typeKey             = "type"
+	configKey           = "config"
 )
 
 // GetControllerProxy retrieves the proxy configuration for the specified controller ID.
-func (d *Database) GetControllerProxy(ctx context.Context, controllerID uint) (proxy *dbmodel.ControllerProxy, err error) {
-	const op = errors.Op("db.GetControllerProxy")
+func (d *Database) GetControllerProxy(ctx context.Context, controllerName string) (_ string, _ map[string]interface{}, err error) {
+	const op = errors.Op("database.GetControllerProxy")
+
 	if err := d.ready(); err != nil {
-		return nil, errors.E(op, err)
+		return "", nil, errors.E(op, err)
 	}
 
 	durationObserver := servermon.DurationObserver(servermon.DBQueryDurationHistogram, string(op))
 	defer durationObserver()
 	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
 
-	db := d.DB.WithContext(ctx)
-
-	proxy = &dbmodel.ControllerProxy{}
-	if err := db.Where("controller_id = ?", controllerID).First(proxy).Error; err != nil {
-		return nil, errors.E(op, dbError(err))
+	secret := dbmodel.NewSecret(controllerProxyKind, controllerName, nil)
+	err = d.GetSecret(ctx, &secret)
+	if errors.ErrorCode(err) == errors.CodeNotFound {
+		return "", nil, nil
+	}
+	if err != nil {
+		zapctx.Error(ctx, "failed to get secret data", zap.Error(err))
+		return "", nil, errors.E(op, err)
 	}
-	return proxy, nil
+	var secretData map[string]interface{}
+	err = json.Unmarshal(secret.Data, &secretData)
+	if err != nil {
+		zapctx.Error(ctx, "failed to unmarshal secret data", zap.Error(err))
+		return "", nil, errors.E(op, err)
+	}
+	typeI, ok := secretData[typeKey]
```

Comment:
Can you put a comment above this line to explain what we're doing regarding extracting the secret data?
Maybe it's just because it's been a while since I looked at this logic but it's a bit confusing.